x-pack/plugin/core/src/main/resources/logs-mappings.json

```diff
@@ -7,6 +7,10 @@
             "type": {
               "type": "constant_keyword",
               "value": "logs"
+            },
+            "@timestamp": {
+              "type": "date",
+              "ignore_malformed": false
```

Comment:
This seems redundant as `data-streams-mappings.json` defines the timestamp

Reply:
Yes, I meant to ask about that, I added the related [check](https://github.com/elastic/elasticsearch/pull/95329/files#diff-4c2bc446708490df863212fbf203e77ed14ec697c2b1ef139c1a4d930916f7caR34), which does seem redundant.
I'll remove

Reply:
The [test failure](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+part-1/32589/console) demonstrated why this was not redundant after all (although misplaced) - [`DataStreamUpgradeRestIT`](https://github.com/elastic/elasticsearch/blob/228fe1f804d93e5a73f88f148c9eb801959e5f10/modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/DataStreamUpgradeRestIT.java#L55) demonstrates a composition of only `logs-settings.json` and `logs-mappings.json`, which requires to make this explicit `@timestamp` exception.
I re-added but as a top-level property